From 0f69dfc7629b2540cdb6667ae03b0f38f618aa4e Mon Sep 17 00:00:00 2001
From: flar2 <asegaert@gmail.com>
Date: Sun, 15 Dec 2013 17:31:17 -0500
Subject: [PATCH 218/276] Revert "Added fsync on/off support."

This reverts commit 51160c4fea7db353de889bc53ab5eef0bf555197.

Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 fs/sync.c |   25 -------------------------
 1 file changed, 25 deletions(-)

diff --git a/fs/sync.c b/fs/sync.c
index f9c190e..0e8db93 100644
--- a/fs/sync.c
+++ b/fs/sync.c
@@ -7,7 +7,6 @@
 #include <linux/fs.h>
 #include <linux/slab.h>
 #include <linux/export.h>
-#include <linux/module.h>
 #include <linux/namei.h>
 #include <linux/sched.h>
 #include <linux/writeback.h>
@@ -18,9 +17,6 @@
 #include <linux/backing-dev.h>
 #include "internal.h"
 
-bool fsync_enabled = false;
-module_param(fsync_enabled, bool, 0755);
-
 #define VALID_FLAGS (SYNC_FILE_RANGE_WAIT_BEFORE|SYNC_FILE_RANGE_WRITE| \
 			SYNC_FILE_RANGE_WAIT_AFTER)
 
@@ -141,9 +137,6 @@ SYSCALL_DEFINE1(syncfs, int, fd)
 	struct super_block *sb;
 	int ret;
 	int fput_needed;
-	
-	if (!fsync_enabled)
-		return 0;
 
 	file = fget_light(fd, &fput_needed);
 	if (!file)
@@ -171,9 +164,6 @@ SYSCALL_DEFINE1(syncfs, int, fd)
  */
 int vfs_fsync_range(struct file *file, loff_t start, loff_t end, int datasync)
 {
-	if (!fsync_enabled)
-		return 0;
-		
 	if (!file->f_op || !file->f_op->fsync)
 		return -EINVAL;
 	return file->f_op->fsync(file, start, end, datasync);
@@ -190,9 +180,6 @@ EXPORT_SYMBOL(vfs_fsync_range);
  */
 int vfs_fsync(struct file *file, int datasync)
 {
-	if (!fsync_enabled)
-		return 0;
-		
 	return vfs_fsync_range(file, 0, LLONG_MAX, datasync);
 }
 EXPORT_SYMBOL(vfs_fsync);
@@ -201,9 +188,6 @@ static int do_fsync(unsigned int fd, int datasync)
 {
 	struct file *file;
 	int ret = -EBADF;
-	
-	if (!fsync_enabled)
-		return 0;
 
 	file = fget(fd);
 	if (file) {
@@ -220,9 +204,6 @@ SYSCALL_DEFINE1(fsync, unsigned int, fd)
 
 SYSCALL_DEFINE1(fdatasync, unsigned int, fd)
 {
-	if (!fsync_enabled)
-		return 0;
-		
 	return do_fsync(fd, 1);
 }
 
@@ -236,9 +217,6 @@ SYSCALL_DEFINE1(fdatasync, unsigned int, fd)
  */
 int generic_write_sync(struct file *file, loff_t pos, loff_t count)
 {
-	if (!fsync_enabled)
-		return 0;
-		
 	if (!(file->f_flags & O_DSYNC) && !IS_SYNC(file->f_mapping->host))
 		return 0;
 	return vfs_fsync_range(file, pos, pos + count - 1,
@@ -303,9 +281,6 @@ SYSCALL_DEFINE(sync_file_range)(int fd, loff_t offset, loff_t nbytes,
 	int fput_needed;
 	umode_t i_mode;
 
-	if (!fsync_enabled)
-		return 0;
-
 	ret = -EINVAL;
 	if (flags & ~VALID_FLAGS)
 		goto out;
-- 
1.7.9.5

